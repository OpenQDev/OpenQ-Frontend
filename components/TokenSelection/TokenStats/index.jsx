/* eslint-disable @next/next/no-img-element */
import React from 'react';
import Image from 'next/image';

const TokenStats = ({ volume, token, pickedNft }) => {
  return (
    <>
      {pickedNft ? (
        <div className='flex gap-2'>
          <img className='w-8 h-8 aspect-square' src={pickedNft.metadata.image} />
          <span className='truncate inline'>
            {pickedNft.name} #{pickedNft.token_id} {pickedNft.metadata.name}
          </span>
        </div>
      ) : (
        <>
          <Image
            width={24}
            className='inline'
            height={24}
            src={token.path || token.logoURI || '/crypto-logos/ERC20.svg'}
            alt='crypto-logo'
          />
          <span>
            {volume} {token.symbol}
          </span>
        </>
      )}
    </>
  );
};

export default TokenStats;
