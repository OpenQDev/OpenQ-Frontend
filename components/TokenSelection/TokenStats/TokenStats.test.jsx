import React from 'react';
import TokenStats from '.';
import renderer from 'react-test-renderer';
import Constants from '../../../test-utils/constant';

describe('TokenStats', () => {
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(<TokenStats volume={'2.8'} token={Constants.tokenMetadata} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
