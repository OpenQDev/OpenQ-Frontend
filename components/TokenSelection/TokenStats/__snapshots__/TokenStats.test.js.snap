// vi Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenStats should match DOM Snapshot 1`] = `
[
  <img
    alt="crypto-logo"
    className="inline"
    data-nimg="1"
    decoding="async"
    height={24}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=https%3A%2F%2Ftokens.1inch.exchange%2F0x6b175474e89094c44da98b954eedeac495271d0f.png&w=48&q=75"
    srcSet="/_next/image?url=https%3A%2F%2Ftokens.1inch.exchange%2F0x6b175474e89094c44da98b954eedeac495271d0f.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Ftokens.1inch.exchange%2F0x6b175474e89094c44da98b954eedeac495271d0f.png&w=48&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={24}
  />,
  <span>
    2.8
     
    DAI
  </span>,
]
`;
